sway/ipc-i3.c

```diff
@@ -0,0 +1,89 @@
+#include <json-c/json.h>
+#include "sway/config.h"
+#include "sway/ipc-json.h"
+#include "sway/ipc-sway.h"
+#include "sway/tree/view.h"
+
+static void ipc_i3_json_describe_view(struct sway_container *c, json_object *object) {
+	const char *app_id = view_get_app_id(c->view);
+	const char *class = app_id;
+	const char *instance = app_id;
+	const char *role = NULL;
+	uint32_t parent_id = 0;
+	uint32_t window_id = 0;
+
+	(void) role;
```

Comment:
I don't think those are necessary.

Reply:
The project doesn't compile if `HAVE_XWAYLAND` isn't defined without those, because the compiler ends up seeing the `if ({role,parent_id,window_id})` and decides that the condition can never be true, and that the variables are actually never used.

Reply:
```c
const char *role = NULL;
if (role) { … }
```

Should compile right?

Reply:
On my system with GCC 8, it doesn't, because GCC thinks (correctly) that role is unused, warns, and stops due to -Werror. I guess you could throw a `volatile` in there, but you'd loose quite more than void-casting or doing a noop on the variables

Reply:
Wrapping all occurrences in ```#ifdef HAVE_XWAYLAND``` should work. Is that bad style?

Reply:
Sure, I can gate these bits with `HAVE_XWAYLAND`, it's not like we have a non-xwayland fallback for those anyway. It might lead to *slightly* larger diffs if we do something about the window_id/parent_id, but I don't think we really have to care about that.